api/token.go

```diff
@@ -234,6 +240,21 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 		return internalServerError(err.Error())
 	}
 
+	if !(config.External.Email.Enabled && config.External.Phone.Enabled) {
```

Comment:
just double checking this is the logic you want?

!(a && b) == (!a || !b)

Reply:
Yeah if both email and phone logins are enabled, we never need to check. So for the inverse cases (either email or phone is disabled or both disabled), we should always check.